test/test-network/systemd-networkd-tests.py

```diff
@@ -2311,5 +2343,26 @@ def test_dhcp_client_route_remove_on_renew(self):
         self.assertRegex(output, f'192.168.5.1 proto dhcp scope link src {address2} metric 1024')
 
 if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--build-dir', help='Path to build dir', dest='build_dir')
+    parser.add_argument('--networkd', help='Path to systemd-networkd', dest='networkd_bin', default=networkd_bin)
+    parser.add_argument('--wait-online', help='Path to systemd-networkd-wait-online', dest='wait_online_bin', default=wait_online_bin)
+    parser.add_argument('--networkctl', help='Path to networkctl', dest='networkctl_bin', default=networkctl_bin)
+    parser.add_argument('--valgrind', help='Enable valgrind', dest='use_valgrind', type=bool, nargs='?', const=True, default=use_valgrind)
+    parser.add_argument('--log-debug', help='Generate debugging logs', dest='enable_debug', type=bool, nargs='?', const=True, default=enable_debug)
+    ns, args = parser.parse_known_args(namespace=unittest)
+
+    if ns.build_dir:
+        networkd_bin = os.path.join(ns.build_dir, 'systemd-networkd')
+        wait_online_bin = os.path.join(ns.build_dir, 'systemd-networkd-wait-online')
+        networkctl_bin = os.path.join(ns.build_dir, 'networkctl')
+    else:
+        networkd_bin = ns.networkd_bin
+        wait_online_bin = ns.wait_online_bin
+        networkctl_bin = ns.networkctl_bin
+    use_valgrind = ns.use_valgrind
+    enable_debug = ns.enable_debug
```

Comment:
<strike>
This seems overly complicated. Also, an explicit argument like `--networkctl` will be ignored if `--build-dir` is specified.
What about the following:
```
parser = argparse.ArgumentParser()
parser.add_argument('--build-dir', help='Path to build dir')
parser.add_argument('--networkd', help='Path to systemd-networkd')
...
</strike>

I"ll submit a separate patch later. I think it'll be quicker then me trying to describe all the changes I'd like to see.